lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnDriverConfigurationGenerator.java

```diff
@@ -0,0 +1,147 @@
+import org.apache.reef.runtime.yarn.driver.YarnDriverRestartConfiguration;
+import org.apache.reef.tang.*;
+import org.apache.reef.tang.annotations.Name;
+import org.apache.reef.tang.annotations.NamedParameter;
+import org.apache.reef.tang.annotations.Parameter;
+import org.apache.reef.tang.exceptions.InjectionException;
+import org.apache.reef.tang.formats.ConfigurationSerializer;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Does client side manipulation of configuration for driver.
+ */
+final class YarnDriverConfigurationGenerator {
+  private static final Logger LOG = Logger.getLogger(YarnDriverConfigurationGenerator.class.getName());
+  private final Set<ConfigurationProvider> configurationProviders;
+  private final int driverRestartEvaluatorRecoverySeconds;
+  private final REEFFileNames fileNames;
+  private final ConfigurationSerializer configurationSerializer;
+
+  @Inject
+  YarnDriverConfigurationGenerator(@Parameter(DriverConfigurationProviders.class)
+                                   final Set<ConfigurationProvider> configurationProviders,
+                                   @Parameter(SubmissionDriverRestartEvaluatorRecoverySeconds.class)
+                                   final int driverRestartEvaluatorRecoverySeconds,
+                                   final ConfigurationSerializer configurationSerializer,
+                                   final REEFFileNames fileNames) {
+    this.configurationProviders = configurationProviders;
+    this.driverRestartEvaluatorRecoverySeconds = driverRestartEvaluatorRecoverySeconds;
+    this.fileNames = fileNames;
+    this.configurationSerializer = configurationSerializer;
+  }
+
+  public Configuration writeConfiguration(final File driverFolder,
+                                          final String jobId,
+                                          final String jobSubmissionFolder) throws IOException {
+    final File driverConfigurationFile = new File(driverFolder, this.fileNames.getDriverConfigurationPath());
+    final Configuration yarnDriverConfiguration = YarnDriverConfiguration.CONF
+        .set(YarnDriverConfiguration.JOB_SUBMISSION_DIRECTORY, jobSubmissionFolder)
+        .set(YarnDriverConfiguration.JOB_IDENTIFIER, jobId)
+        .set(YarnDriverConfiguration.CLIENT_REMOTE_IDENTIFIER, ClientRemoteIdentifier.NONE)
+        .set(YarnDriverConfiguration.JVM_HEAP_SLACK, 0.0)
+        .build();
+
+    final ConfigurationBuilder configurationBuilder = Tang.Factory.getTang().newConfigurationBuilder();
+    for (final ConfigurationProvider configurationProvider : this.configurationProviders) {
+      configurationBuilder.addConfiguration(configurationProvider.getConfiguration());
+    }
+    final Configuration providedConfigurations =  configurationBuilder.build();
+
+    Configuration driverConfiguration = Configurations.merge(
```

Comment:
Kind of yes. But these configurations could also be added to the final `.merge()` call, right?